pkg/internal/parser.go

```diff
@@ -162,37 +152,276 @@ func sanitize(groupName string) string {
+					a.InputType = fmt.Sprintf("*%sapplyconfig.%sApplyConfiguration", strings.ToLower(name), name)
+				}
+			} else {
+				a.InputType = fmt.Sprintf("*%sapi%s.%s", a.PkgName, a.Version, name)
+				if *extension.Verb == "apply" {
+					a.InputType = fmt.Sprintf("*%sapply%s.%sApplyConfiguration", a.PkgName, a.Version, name)
+				}
+			}
+			a.InputName = strings.ToLower(name)
+		}
+
+		if *extension.Verb == "get" {
+			a.Method = *extension.Method
+			if err := templateExecute(getTemplate, *a); err != nil {
+				return err
+			}
+		}
+
+		if *extension.Verb == "create" {
+			a.Method = *extension.Method
+			adjTemplate := adjustTemplate(createTemplate, "create")
+			if err := templateExecute(adjTemplate, *a); err != nil {
+				return err
+			}
+		}
+
+		if *extension.Verb == "update" {
+			a.Method = *extension.Method
+			adjTemplate := adjustTemplate(updateTemplate, "update")
+			if err := templateExecute(adjTemplate, *a); err != nil {
+				return err
+			}
+		}
+
+		if *extension.Verb == "apply" {
+			a.Method = *extension.Method
+			adjTemplate := adjustTemplate(applyTemplate, "apply")
+			if err := templateExecute(adjTemplate, *a); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Execute template and append the contents to the writer.
+func templateExecute(templateName string, data api) error {
+	funcMap := template.FuncMap{
+		"value": util.Value,
+		"typepkg": func(a api) string {
+			return fmt.Sprintf("*%sapi%s.%s", a.PkgName, a.Version, a.Name)
+		},
+	}
+	templ, err := template.New("wrapper").Funcs(funcMap).Parse(templateName)
 	if err != nil {
 		return err
 	}
-	return templ.Execute(a.writer, a)
+	return templ.Execute(data.writer, data)
+}
+
+// hasVerb returns true if the verb is to be scaffolded,
+// if it is a part of "skipVerbs" then it returns false.
+func (a *api) hasVerb(verb string) bool {
+	if len(a.SkipVerbs) == 0 {
+		return true
+	}
+	for _, s := range a.SkipVerbs {
+		if verb == s {
+			return false
+		}
+	}
+	return true
+}
+
+// getPkgType returns the result override package path and the type.
+func getPkgType(input string) (string, string) {
+	parts := strings.Split(input, ".")
+	return parts[len(parts)-1], strings.Join(parts[0:len(parts)-1], ".")
+}
+
+// appendImports adds the regular imports needed for scaffolding.
+func (p *packages) appendGenericImports(importList []string) []string {
+	return append(importList, "\"context\"", "\"fmt\"", "\"k8s.io/apimachinery/pkg/types\"", "\"k8s.io/client-go/rest\"",
+		"\"github.com/kcp-dev/logicalcluster\"", util.ImportFormat("metav1", "k8s.io/apimachinery/pkg/apis/meta/v1"), "\"k8s.io/apimachinery/pkg/watch\"", util.ImportFormat("kcp", "github.com/kcp-dev/apimachinery/pkg/client"),
+		util.ImportFormat(fmt.Sprintf("%sapi%s", p.Name, p.Version), p.APIPath),
+		util.ImportFormat(fmt.Sprintf("%s%s", p.Name, p.Version), fmt.Sprintf("%s/typed/%s/%s", p.ClientPath, p.Name, p.Version)))
+}
+
+func adjustTemplate(template, verb string) string {
```

Comment:
Why did we choose to adjust the template instead of writing a template that handles both cases?

Reply:
Its just because both the templates are similar (one is based out of another - in the sense if the user wants to override a `Create` template with `CreateScale`, the parameters will be similar). We could have it separately too, just that the templating logic will still be the same. If the values to be overriden are not provided we would have to get it from the base template.
But I completely agree that the templating logic which we have followed is pretty convoluted. A simple way would have been to pass inputs as `map[string]interface{}` instead of struct which is making the code look really clumsy. Have created an issue here for cleanup #18

Reply:
the conditionals can be in the template itself, no?